sunkit_image/flct/tests/test_flct.py

```diff
@@ -0,0 +1,303 @@
+
+    assert str(record.value) == "The absolute value of 'xoff' and 'yoff' must be less than skip."
+
+    with pytest.raises(ValueError) as record:
+        _ = flct.flct(images[0], images[1], 1, 1, 5, kr=40)
+
+    assert str(record.value) == "The value of 'kr' must be between 0 and 20."
+
+    with pytest.raises(ValueError) as record:
+        _ = flct.flct(images[0], images[1], 1, 1, 5, kr=0.5, skip=1000)
+
+    assert str(record.value) == "Skip is greater than the input dimensions"
+
+
+def test_flct_dat(images_dat, outputs_dat):
+
+    vx, vy, vm = flct.flct(images_dat[0], images_dat[1], 1, 1, 5, kr=0.5)
+
+    assert np.allclose(vx, outputs_dat[0])
+    assert np.allclose(vy, outputs_dat[1])
+    assert np.allclose(vm, outputs_dat[2])
+
+    # Errors checks
+    order = "random"
+
+    with pytest.raises(ValueError) as record:
+        _ = flct.flct(images_dat[0], images_dat[1], 1, 1, 5, order, kr=0.5)
+
+    assert (
+        str(record.value)
+        == "The order of the arrays is not correctly specified. It can only be 'row' or 'column'"
+    )
+
+    with pytest.raises(ValueError) as record:
+        _ = flct.flct(images_dat[0], images_dat[1], 1, 1, 5, kr=0.5, skip=-1)
+
+    assert str(record.value) == "Skip value must be greater than zero."
+
+    with pytest.raises(ValueError) as record:
+        _ = flct.flct(images_dat[0], images_dat[1], 1, 1, 5, kr=0.5, skip=1, xoff=4)
+
+    assert str(record.value) == "The absolute value of 'xoff' and 'yoff' must be less than skip."
+
+    with pytest.raises(ValueError) as record:
+        _ = flct.flct(images_dat[0], images_dat[1], 1, 1, 5, kr=40)
+
+    assert str(record.value) == "The value of 'kr' must be between 0 and 20."
+
+    with pytest.raises(ValueError) as record:
+        _ = flct.flct(images_dat[0], images_dat[1], 1, 1, 5, kr=0.5, skip=1000)
+
+    assert str(record.value) == "Skip is greater than the input dimensions"
+
+
+# These tests are dummy tests. They are written just to make sure that FLCT runs on optional
+# parameters also. We did not have any values to compare our results against so this is why
+# these tests are at the end such that only after all the valid tests are passed then these
+# are executed. These are not tests in the strictest sense rather it is designed to increase
+# the test coverage for lines containing the setting of optional arguments.
+
+
+def test_flct_optional(images_dat):
+    _ = flct.flct(
+        images_dat[0],
+        images_dat[1],
+        1,
+        1,
+        4,
+        skip=4,
+        interp=True,
+        quiet=True,
+        absflag=True,
+        biascor=True,
+        pc=True,
+        xoff=-2,
+        yoff=-2,
+    )
+
+    _ = flct.flct(
+        images_dat[0],
+        images_dat[1],
+        1,
+        1,
+        0,
+        interp=True,
+        quiet=True,
+        absflag=True,
+        biascor=True,
+        pc=True,
+    )
+    assert True
```

Comment:
Then if you move the some of the functions to flct/util, you should move those tests to a test_flcy_util fie.